aim/sdk/adapters/lightgbm.py

```diff
@@ -0,0 +1,97 @@
+from typing import Optional
+
+from aim import Run
+from aim.ext.resource import DEFAULT_SYSTEM_TRACKING_INT
+
+try:
+    from lightgbm.callback import CallbackEnv
+except ImportError:
+    raise RuntimeError(
+        'This contrib module requires Lightgbm to be installed. '
+        'Please install it with command: \n pip install lightgbm'
+    )
+
+
+class AimCallback:
+    """
+    Records evaluation data into Aim.
+    Similar to `lightgbm.record_evaluation` callback:
+    https://lightgbm.readthedocs.io/en/latest/pythonapi/lightgbm.record_evaluation.html
+    """
+
+    def __init__(self, repo: Optional[str] = None,
+                 experiment: Optional[str] = None,
+                 system_tracking_interval: Optional[int]
+                 = DEFAULT_SYSTEM_TRACKING_INT,
+                 log_system_params: Optional[int] = True):
+        self._repo_path = repo
+        self._experiment = experiment
+        self._system_tracking_interval = system_tracking_interval
+        self._log_system_params = log_system_params
+        self._run = None
+        self._run_hash = None
+
+        # callback parameters
+        self.order = 15
```

Comment:
what `order` stands for?

Reply:
I cannot really say, its not documented and all the `LightGBM`'s callbacks have order set in them, I picked a value in between

Reply:
@mahnerak any idea here?

Reply:
When there are multiple Callbacks registered this defines the precedence of their calls.
I think this value should be as great (as late) as possible, but before checkpointing.

It looks like `30` is for early stopping and serialization of the model, and `20` for recording some logs. So something in between will work better.

Reply:
@devfox-se please double-check this.